framework/routing/__init__.py

```diff
@@ -102,9 +102,10 @@ def wrap_with_renderer(fn, renderer, renderer_kwargs=None, debug_mode=True):
     """
     @functools.wraps(fn)
     def wrapped(*args, **kwargs):
-        if session:
-            session_error_code = session.data.get('auth_error_code')
-        else:
+        current_session = get_session()
+        try:
+            session_error_code = current_session['auth_error_code']
+        except KeyError:
             session_error_code = None
```

Comment:
Comment for myself: no need to use `try/except` if we use `.get()`; in addition, revert back to use `session` instead of calling `get_session()`.

```python
session_error_code = session.get('auth_error_code', None) if session else None
```